shift-array/shift-array.py

```diff
@@ -6,68 +6,56 @@
 input2 = [4, 8, 15, 23, 42]
 inputValue2 = 16
 
-# declarations
-# length = 0
-# middle = 0
+# determine length of input list
 
 
-# determine length of input list
+def new_list(input_list, input_value):
 
-# def list_length(input_list):
-#     length = 0
-#     for item in input_list:
-#         length += 1
-#     print('length', length)
-#     return length
+    length = 0
+
+    for item in input_list:
+        length += 1
+    # print('length', length)
 
 
 # find middle position of array
 
-def middle_position(length):
-    print(length)
     middle = 0  
     if length % 2 == 0:
         middle = length / 2
-        print('middle even', middle)
-        return middle
+        # print('middle even', middle)   
     else:
         middle = (length // 2) + 1
-        print('middle odd', middle)
-        return middle
+        # print('middle odd', middle)
 
 
 # for loop through each index and copy value into new list
 
-def new_list(input_list, input_value):
-
-    length = 0
-
-    for item in input_list:
-        length += 1
-    print('length', length)
-    
-    middle = 0  
-    if length % 2 == 0:
-        middle = length / 2
-        print('middle even', middle)   
-    else:
-        middle = (length // 2) + 1
-        print('middle odd', middle)
-        
     output_list = []
+    counter = 0
     for index in input_list:
-        if index < middle:
+        
+        # print('counter %d' % counter)
+        # print('index %d' % index)
+        if counter < middle:
             output_list += [index]
-        elif index == middle:
+            counter += 1
+        elif counter == middle:
             output_list += [input_value]
             output_list += [index]
+            counter += 1
         else:
             output_list += [index]
+            counter += 1
```

Comment:
You have duplicate code in these conditionals. Try and keep you code DRY by extracting that out.